docker-compose-rule-core/src/main/java/com/palantir/docker/compose/execution/DefaultDockerCompose.java

```diff
@@ -244,9 +263,20 @@ private static boolean downCommandWasPresent(String output) {
         return !output.contains("No such command");
     }
 
-    private String psOutput(String service) throws IOException, InterruptedException {
-        String psOutput = command.execute(Command.throwingOnError(), "ps", service);
-        validState(!Strings.isNullOrEmpty(psOutput), "No container with name '" + service + "' found");
-        return psOutput;
+    private static ErrorHandler swallowingServiceDoesNotExist() {
+        return (exitCode, output, commandName, commands) -> {
+            if (exitCode == 1 && output.isEmpty()) {
```

Comment:
Why exit code 1 and not all exit codes? Will this silently fail if another exit code is thrown?

Reply:
Yeah this is pretty specific, its really only trying to add debug information for when docker can't find a service 
```
~/code/foundry-academy-backend
$ docker-compose ps some_service
ERROR: No such service: some_service
~/code/foundry-academy-backend
$ echo $?
1
```

Seems it would be more proper for this function to check for various failure exit codes and throw an exception explicitly, while still providing output with suggestions around what the potential failure mode was?